### PR TITLE
Cope with extra files in Godeps/ directory

### PR DESCRIPTION
**Pull request by sttts**
Fixes issues due to https://github.com/kubernetes/kubernetes/pull/59587#issuecomment-368575420.

### CODE REVIEW

**Comment by sbezverk** [comment]
@sttts Thanks for offline explanation. The proposed solution, I think should help.

File: artifacts/scripts/util.sh [hunk 1]
```diff
@@ -757,6 +757,11 @@ function update_full_godeps() {
     echo "Running godep save."
     godep save ./...
 
+    # restore all other files of Godeps/ (like OWNERS), but preserve the new Godeps.json
+    cp Godeps/Godeps.json Godeps/Godeps.json.preserve
```
Comment (cblecker) [inline artifacts/scripts/util.sh]: So this basically moves the Godeps.json to an untracked file, then re-checks out the dir from HEAD, then moves the untracked file back overtop the correct filename? I haven't dug too much into the publisher code.

Comment (sttts) [reply artifacts/scripts/util.sh]: yes. Godeps does not like the have an existing Godeps/ dir. So we remove it. But then we lost Godeps/OWNERS. Any following commit changing OWNERS will break (if we don't preserve it).

Comment (cblecker) [reply artifacts/scripts/util.sh]: Right. Okay, this should work then. A very duct tape solution, but 🤞 we won't have to deal with Godeps too much longer.

Comment (sttts) [reply artifacts/scripts/util.sh]: The scripts is 50% duct-tape around godep.

**Pull request merged**